### ISSUE TITLE
vue/no-unused-properties does not detect usages in watch handlers

### ISSUE THREAD

**Issue by scp-mb** [vuejs/eslint-plugin-vue#1245]
**Tell us about your environment**
* **ESLint version :6.8.0** 
* **eslint-plugin-vue version: 7.0.0-alpha.9** 
* **Node version:11.5.0** 

**Please show your full configuration:**

```json
module.exports = {
  env: {
    node: true
  },
  extends: [
    'plugin:vue/essential',
    '@vue/standard'
  ],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: 'babel-eslint'
  },
  root: true,
  rules: {
    /* omitted other rules */
    'vue/no-unused-properties': [ 'warn', {
      'groups': ['props', 'data', 'computed', 'methods', 'setup']
    }]
  }
}

```

**What did you do?**	

Calls to methods within watch handlers do not get picked up as being used by vue/no-unused-properties. GetMoreData will be highlighted as unused
```vue
watch: {
  someData: {
    handler: 'GetMoreData',
    immediate: true
  }
},

......


methods: {
  GetMoreData () {
    /* business logic here */
  }
}
```

**What did you expect to happen?**
GetMoreData is not flagged as unused

**What actually happened?**

`'GetMoreData' of method found, but never used. eslint(vue/no-unused-properties)`

**Comment by ota-meshi** [comment]
Thank you for this issue!

It looks like a bug.
